pkg/config/initoptions.go

```diff
@@ -142,19 +144,24 @@ func calcPackageNamespace(packageDir string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	// Return the first non-empty namespace found. Cluster-scoped
+	// Return the non-empty unique namespace if found. Cluster-scoped
 	// resources do not have namespace set.
+	currentNamespace := metav1.NamespaceDefault
 	for _, node := range nodes {
 		rm, err := node.GetMeta()
-		if err != nil {
+		if err != nil || len(rm.ObjectMeta.Namespace) <= 0 {
 			continue
 		}
-		if len(rm.ObjectMeta.Namespace) > 0 {
-			return rm.ObjectMeta.Namespace, nil
+		if currentNamespace == metav1.NamespaceDefault {
+			currentNamespace = rm.ObjectMeta.Namespace
```

Comment:
This doesn't seem right. If the namespace is set to `default` here, we can't just ignore it.

Reply:
change to use metav1.NamespceNone. It's in effect the same. currentNamespace is set(initialized) to the namespace of the first node. And in the further loops it only compares if it is same to that. If there happen to be a namespace called default or none, that's fine as it is the same. It only matter when it find a different one than the currentNamespace.

Reply:
So I think if we find any resource that doesn't have the namespace set, we should ask the user to specify the namespace. In these situations, we don't really know where the user actually wants the inventory to end up.

Reply:
Got it, return error instead for this case.